### PR TITLE
Support Release build type for Apple builds

### PR DESCRIPTION
**Pull request by nhachicha**
Also enabling Proguard/shrinker for Android
fixes https://github.com/realm/realm-kotlin/issues/293 and https://github.com/realm/realm-kotlin/issues/142

### CODE REVIEW

File: packages/cinterop/build.gradle.kts [hunk 1]
```diff
@@ -277,74 +310,102 @@ val capiAndroidRelease by tasks.registering {
...
 }
 
 fun Task.build_C_API_Android(releaseBuild: Boolean = false) {
     val buildType = if (releaseBuild) "Release" else "Debug"
     doLast {
         // x86_64
         exec {
-            workingDir(project.file("$relativeCorePath"))
-            commandLine("tools/cross_compile.sh", "-t", buildType, "-a", "x86_64", "-o", "android", "-f", "-DREALM_ENABLE_SYNC=0 -DREALM_NO_TESTS=ON")
+            workingDir(project.file("$absoluteCorePath"))
```
Comment (cmelchior) [inline packages/cinterop/build.gradle.kts:329]: Are we even using these tasks for anything? Shouldn't the CMake file handle all of this?

**Review by cmelchior — APPROVED** [review]

File: packages/cinterop/build.gradle.kts (hunk 1 above)
Comment (nhachicha) [reply packages/cinterop/build.gradle.kts:329]: These tasks are using CMake with various toolchains, this also gives us the granularity to control some flags (like `-UseModernBuildSystem=NO` more easily)

File: packages/cinterop/build.gradle.kts [hunk 2]
```diff
@@ -88,24 +88,53 @@ android {
...
 }
 
-val nativeLibraryIncludesMacosUniversal = listOf(
-    "-include-binary", "$cinteropCorePath/capi_macos_universal/src/realm/object-store/c_api/librealm-ffi-static-dbg.a",
-    "-include-binary", "$cinteropCorePath/capi_macos_universal/src/realm/librealm-dbg.a",
-    "-include-binary", "$cinteropCorePath/capi_macos_universal/src/realm/parser/librealm-parser-dbg.a",
-    "-include-binary", "$cinteropCorePath/capi_macos_universal/src/realm/object-store/librealm-object-store-dbg.a"
+val nativeLibraryIncludesMacosUniversalRelease = listOf(
+    "-include-binary",
+    "$absoluteCorePath/capi_macos_universal/src/realm/object-store/c_api/librealm-ffi-static.a",
```
Comment (rorbech) [inline packages/cinterop/build.gradle.kts:93]: Really annoying that the directory `capi_macos_universal` doesn't follow the pattern of all the other build directories by prefixing `build_`

File: packages/cinterop/build.gradle.kts [hunk 3]
```diff
@@ -88,24 +88,53 @@ android {
     }
 }
 
-val nativeLibraryIncludesMacosUniversal = listOf(
-    "-include-binary", "$cinteropCorePath/capi_macos_universal/src/realm/object-store/c_api/librealm-ffi-static-dbg.a",
-    "-include-binary", "$cinteropCorePath/capi_macos_universal/src/realm/librealm-dbg.a",
-    "-include-binary", "$cinteropCorePath/capi_macos_universal/src/realm/parser/librealm-parser-dbg.a",
-    "-include-binary", "$cinteropCorePath/capi_macos_universal/src/realm/object-store/librealm-object-store-dbg.a"
+val nativeLibraryIncludesMacosUniversalRelease = listOf(
```
Comment (rorbech) [inline packages/cinterop/build.gradle.kts:91]: Extremely verbose. I guess there are some veriations, but I guess we could maybe define the libs 
```
val libs = setOf("lib/librealm-ffi-static.a"....)
val libsDbg = setOf("lib/librealm-ffi-static.a"....)
```
And define the compiler arguments by something from these, by some mapping
```
val nativeLibraryIncludesIosArm64Debug = libsDbg.map { ... }
```

File: packages/cinterop/build.gradle.kts [hunk 4]
```diff
@@ -131,7 +160,7 @@ kotlin {
             cinterops.create("realm_wrapper") {
                 defFile = project.file("src/nativeCommon/realm.def")
                 packageName = "realm_wrapper"
-                includeDirs(project.file("$relativeCorePath/src/"))
+                includeDirs("$absoluteCorePath/src/")
```
Comment (rorbech) [inline packages/cinterop/build.gradle.kts:163]: Have you verified that this still allows compilation from both inside `packages` and `test`. As far as I recall there was some issue with that.

File: packages/cinterop/build.gradle.kts (hunk 1 above)
Comment (rorbech) [reply packages/cinterop/build.gradle.kts:329]: I don't think these tasks are triggered anymore after building through the monorepo Cmake-infrastructure, so you would need to incorporate stuff into the `CMakeLists.txt`. At least I removed them in the `frozen-architecture`-branch without causing issues.

**Review by rorbech — APPROVED** [review]
LGTM, but please verify that we can build/test things both from `packages` and `test` after updating the core paths.

**Pull request merged**